CMakeLists.txt

```diff
@@ -187,7 +187,7 @@ endif()
 if(BUILD_SHARED_LIBS)
 	target_compile_definitions(sentry PRIVATE SENTRY_BUILD_SHARED)
 else()
-	target_compile_definitions(sentry PUBLIC SENTRY_BUILD_STATIC)
+	target_compile_definitions(sentry PRIVATE SENTRY_BUILD_STATIC)
```

Comment:
You might want to keep this option public.
On Windows, this definition is used to set `__declspec(dllimport)` if it is shared and `extern` if it is static.

Reply:
That is a good point

Reply:
> You might want to keep this option public.
> On Windows, this definition is used to set `__declspec(dllimport)` if it is shared and `extern` if it is static.

Unless I understand `PUBLIC` / `PRIVATE` wrong ; as long as `sentry.h` (which is the only file that I found that uses `SENTRY_BUILD_STATIC`) is in sentry's `target_sources`, `PRIVATE `is enough and will produce the expected behavior.

Reply:
> Unless I understand `PUBLIC` / `PRIVATE` wrong ; as long as `sentry.h` (which is the only file that I found that uses `SENTRY_BUILD_STATIC`) is in sentry's `target_sources`, `PRIVATE `is enough and will produce the expected behavior.

It's the combination that is important. When a dependency includes `senty.h`, it needs to know whether it is a `shared` or `static` sentry, that is in use.
You can also `generate` a header that will automagicaly set the correct defines.
See [GenerateExportHeader](https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html).

`target_sources` is a way to add sources to the target itself and/or to its dependencies.